### DISCUSSION TITLE
CreateDefaultSubobject<T> 함수는 뭐하는 놈이죠?

### DISCUSSION THREAD

**Discussion by simeddk** [simeddk/Tore04_BasicSyntaxCPP#3] [Q&A] [answered]
이거 BeginPlay에서 써도 되는 함수 맞죠?
고수님들 도와주세요!
`NewObject<T>` ... 이런 함수는 또 뭔가요? 머리가 너무 아파요ㅠㅠ

**Comment by HyeonLang** [comment]
김재우
CreateDefaultSubobject<T>는 생성자에서만 호출될 수 있습니다.
NewObject<T>를 BeginPlay에서 호출해서 같은 역할을 할 수 있습니다.

**Comment by ehddnjs1200** [comment, marked as answer]
정동원

CreateDefaultSubobject<T> 함수는 Unreal Engine에서 주로 클래스의 생성자(constructor) 내에서 사용되며 BeginPlay 함수에서 사용하는 것은 적절하지 않습니다. 
그 이유는 CreateDefaultSubobject<T>는 오브젝트가 생성될 때 기본적으로 서브 오브젝트를 초기화하는 데 사용됩니다.
따라서 오브젝트가 이미 생성되고, 월드에 배치된 후에 호출되는 BeginPlay 함수에서 사용되는 것은 적절하지 않습니다.
NewObject<T> 런타임 중에 새로운 오브젝트를 생성하는 데 사용되는 함수입니다.
